hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1060,6 +1074,44 @@ public void updateMembers(Collection<MemberInfo> members) {
         }
     }
 
+    private boolean shouldProcessMemberUpdate(Map<Address, MemberImpl> currentMembers,
+                                              Collection<MemberInfo> newMemberInfos) {
+
+        int currentMembersSize = currentMembers.size();
+        int newMembersSize = newMemberInfos.size();
+
+        if (currentMembersSize > newMembersSize) {
+            logger.severe("Received an older member update, no need to process...");
+            return false;
+        }
+
+        // member-update process only accepts new member updates
+        if (currentMembersSize == newMembersSize) {
+            Set<MemberInfo> currentMemberInfos = createMemberInfoSet(currentMembers.values());
+            if (currentMemberInfos.containsAll(newMemberInfos)) {
+                logger.finest("Received a periodic member update, no need to process...");
+            } else {
+                logger.warning("Received an inconsistent member update "
+                        + "which contains new members and removes some of the current members! "
+                        + "Ignoring and requesting a new member update...");
+                nodeEngine.getOperationService().send(new TriggerMemberListPublishOperation(), getMasterAddress());
```

Comment:
Your scenario cannot happen at the same since the operations done under the cluster service lock. If the member is dead, we remove that and sent the event to other members then join starts. So I assume that event will arrive first then if the inconsistency happens we request member-list again. I cannot foresee the case that it will hog the network.